lib/core/hard_machine.rb

```diff
@@ -0,0 +1,61 @@
+require "core/board"
+require "core/player"
+
+# HardMachine class follows the minmax algorithm to choose the next move.
+# The algorithm has been implemented taking into account depth in order
+# to be able to identify quicker wins over longer ones.
+class HardMachine < Player
+  INITIAL_DEPTH = 0
+
+  def initialize(symbol)
+    super(symbol)
+    @opponent = Player.new(symbol == "X" ? "O" : "X")
```

Comment:
Hmm, not sure this logic should be here. It feels weird that your computer player needs to change if you change a UI concern like the mark that's displayed for the player.